### PR TITLE
cc3220 port

### PR DESCRIPTION
**Pull request by balazsracz**
Ports the CC32xx sockets/wifi stack to the CC3220SDK and adds example applications and boards for the CC3220SF, albeit running from RAM at the moment.

### CODE REVIEW

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/config.hxx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/// - Software version
+///
+/// This data will be used for all purposes of the identification:
+///
+/// - the generated cdi.xml will include this data
+/// - the Simple Node Ident Info Protocol will return this data
+/// - the ACDI memory space will contain this data.
+extern const SimpleNodeStaticValues SNIP_STATIC_DATA = {
+    4,               "OpenMRN", "Test IO Board - Tiva Launchpad 123",
+    "ek-tm4c123gxl", "1.01"};
```
Comment (bakerstu) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/config.hxx]: Wrong SNIP data to describe this node.

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file main.cxx
+ *
+ * Main file for the io board application on the Tiva Launchpad board.
```
Comment (bakerstu) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx]: Incorrect board described.

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    // physical CAN port and the USB port will make this firmware act as an
+    // USB-CAN adapter in addition to the producers/consumers created above.
+    //
+    // If a port is enabled, it must be functional or else the stack will
+    // freeze waiting for that port to send the packets out.
+#if defined(HAVE_PHYSICAL_CAN_PORT)
+    stack.add_can_port_select("/dev/can0");
+#endif
+#if defined(SNIFF_ON_USB)
+    stack.add_gridconnect_port("/dev/serUSB0");
```
Comment (bakerstu) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx]: Do we really need this option, the CC32x platform does not have native USB.

File: src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c [hunk 4]
```diff
@@ -0,0 +1,66 @@
+/** \copyright
+ * Copyright (c) 2012, Stuart W Baker
```
Comment (bakerstu) [inline src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c]: Should this be your copyright?

File: src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+#include <ti/drivers/dpl/SemaphoreP.h>
+
+#include <FreeRTOS.h>
+#include <semphr.h>
+#include <queue.h>
+
+/*
+ *  ======== SemaphoreP_pend ========
+ */
+SemaphoreP_Status SemaphoreP_pend(SemaphoreP_Handle handle, uint32_t timeout)
```
Comment (bakerstu) [inline src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c]: Should we pull the implementation out of the SimpleLink SDK instead?

File: boards/ti-cc3220sf-launchpad/HwInit.cxx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file HwInit.cxx
+ * This file represents the hardware initialization for the TI Tiva MCU.
```
Comment (bakerstu) [inline boards/ti-cc3220sf-launchpad/HwInit.cxx]: Incorrect MCU name.

File: src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c (hunk 5 above)
Comment (balazsracz) [inline src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c]: I'm not sure what you mean. The SDK implementation is broken, that's why we're patching it here.

Comment (bakerstu) [inline src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c]: Ah, right, this is the bug.

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/config.hxx (hunk 1 above)
Comment (balazsracz) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/config.hxx]: done

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx (hunk 2 above)
Comment (balazsracz) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx]: done

File: applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx (hunk 3 above)
Comment (balazsracz) [inline applications/io_board/targets/freertos.armv7m.ti-launchpad-cc3220sf/main.cxx]: done

File: src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c (hunk 4 above)
Comment (balazsracz) [inline src/freertos_drivers/net_cc32xx/CC3220/semaphore_freertos_patch.c]: done

File: boards/ti-cc3220sf-launchpad/HwInit.cxx (hunk 6 above)
Comment (balazsracz) [inline boards/ti-cc3220sf-launchpad/HwInit.cxx]: done

**Pull request merged**